libsolidity/ast/Types.cpp

```diff
@@ -841,13 +841,14 @@ TypePointer RationalNumberType::forLiteral(Literal const& _literal)
 		TypePointer compatibleBytesType;
 		if (_literal.isHexNumber())
 		{
+			ASTString const valueString = _literal.valueWithoutUnderscores();
 			size_t digitCount = count_if(
```

Comment:
Why is this still needed?

Reply:
Well, we're catching two bugs here. First (a few lines below, the original report), where too large literals where wrongly getting the type FixedBytes-type with byte-count > 32), and secondly that the counting is wrong if we don't strip the underscores from the literal.

Reply:
But we DO strip the underscores, so no need to count in such a way, just `digitCount = valueString.length() - 2` should be enough, shouldn't it?

Reply:
ah! okay, now I get you. you are right indeed.